io.openems.edge.meter.siemens.pac1600/src/io/openems/edge/meter/siemens/pac1600/MeterSiemensPAC1600Impl.java

```diff
@@ -0,0 +1,166 @@
+	protected ConfigurationAdmin cm;
+
+
+	public MeterSiemensPAC1600Impl() {
+		super(//
+				OpenemsComponent.ChannelId.values(), //
+				ModbusComponent.ChannelId.values(), //
+				ElectricityMeter.ChannelId.values(), //
+				MeterSiemensPAC1600.ChannelId.values() //
+		);
+	}
+
+	@Reference(policy = ReferencePolicy.STATIC, policyOption = ReferencePolicyOption.GREEDY, cardinality = ReferenceCardinality.MANDATORY)
+	protected void setModbus(BridgeModbus modbus) {
+		super.setModbus(modbus);
+	}
+
+	@Activate
+	void activate(ComponentContext context, Config config) throws OpenemsException {
+		this.meterType = config.type();
+
+		if (super.activate(context, config.id(), config.alias(), config.enabled(), config.modbusUnitId(), this.cm,
+				"Modbus", config.modbus_id())) {
+			return;
+		}
+		// this.config = config;
+	}
+
+	@Deactivate
+	protected void deactivate() {
+		super.deactivate();
+	}
+
+	@Override
+	public MeterType getMeterType() {
+		return this.meterType;
+		// return this.config.type();
+	}
+
+	@Override
+	protected ModbusProtocol defineModbusProtocol() throws OpenemsException {
+		ModbusProtocol modbusProtocol = new ModbusProtocol(this, //
+				new FC3ReadRegistersTask(1, Priority.HIGH, //
+					m(ElectricityMeter.ChannelId.VOLTAGE_L1, new UnsignedDoublewordElement(1),ElementToChannelConverter.SCALE_FACTOR_1),
+					m(ElectricityMeter.ChannelId.VOLTAGE_L2, new UnsignedDoublewordElement(3),ElementToChannelConverter.SCALE_FACTOR_1),
+					m(ElectricityMeter.ChannelId.VOLTAGE_L3, new UnsignedDoublewordElement(5),ElementToChannelConverter.SCALE_FACTOR_1),
+					
+					m(ElectricityMeter.ChannelId.CURRENT_L1, new UnsignedDoublewordElement(7),ElementToChannelConverter.SCALE_FACTOR_MINUS_1),
+					m(ElectricityMeter.ChannelId.CURRENT_L2, new UnsignedDoublewordElement(9),ElementToChannelConverter.SCALE_FACTOR_MINUS_1),
+					m(ElectricityMeter.ChannelId.CURRENT_L3, new UnsignedDoublewordElement(11),ElementToChannelConverter.SCALE_FACTOR_MINUS_1),
+					
+					new DummyRegisterElement(13, 18), //
+					
+					m(ElectricityMeter.ChannelId.ACTIVE_POWER_L1, new SignedDoublewordElement(19),ElementToChannelConverter.SCALE_FACTOR_MINUS_2),
+					m(ElectricityMeter.ChannelId.ACTIVE_POWER_L2, new SignedDoublewordElement(21),ElementToChannelConverter.SCALE_FACTOR_MINUS_2),
+					m(ElectricityMeter.ChannelId.ACTIVE_POWER_L3, new SignedDoublewordElement(23),ElementToChannelConverter.SCALE_FACTOR_MINUS_2),
+					
+					m(ElectricityMeter.ChannelId.REACTIVE_POWER_L1, new SignedDoublewordElement(25),ElementToChannelConverter.SCALE_FACTOR_MINUS_2),
+					m(ElectricityMeter.ChannelId.REACTIVE_POWER_L2, new SignedDoublewordElement(27),ElementToChannelConverter.SCALE_FACTOR_MINUS_2),
+					m(ElectricityMeter.ChannelId.REACTIVE_POWER_L3, new SignedDoublewordElement(29),ElementToChannelConverter.SCALE_FACTOR_MINUS_2)),
+				
+				new FC3ReadRegistersTask(49, Priority.HIGH, // DummyRegisterElement didn`t work here, there are only 30 Registers in one Task allowed
```

Comment:
Normally the maximum length of a task is much bigger than 30, it's 126. I think the reason that `DummyRegisterElement` didn't work here is because the registers are empty. If they were filled with something like FFFF, it may work I think.